### ISSUE TITLE
Buy bot post buy order before auction starts

### ISSUE THREAD

**Issue by ben-kaufman** [gnosis/dx-services#22]
I am currently having a problem with the buy bot when running the bots. The buy bot is trying to post buy order on an auction which didn't start yet (when its state is WAITING_FOR_AUCTION_TO_START.
That's probably because the price of the sale is lower than the market price and the bot does not verify that the auction started.
Here's the error message for more details:
```
{ AssertionError [ERR_ASSERTION]: The auction hasn't started yet
    at AuctionRepoImpl.postBuyOrder (/usr/src/app/src/repositories/AuctionRepo/AuctionRepoImpl.js:867:5)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '==' }
{ AssertionError [ERR_ASSERTION]: The auction hasn't started yet
    at AuctionRepoImpl.postBuyOrder (/usr/src/app/src/repositories/AuctionRepo/AuctionRepoImpl.js:867:5)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '==' }
```

**Comment by dasanra** [comment]
This issue is duplicated and related to this one

https://github.com/gnosis/dx-services/issues/10

**Comment by dasanra** [comment]
We have added a check to this state to show a more convenient message. Will be available from v2.0.3:

https://github.com/gnosis/dx-services/pull/50
